translator/trace/jaeger/protospan_to_jaegerthrift.go

```diff
@@ -136,18 +136,21 @@ func ocNodeAndResourceToJaegerProcess(node *commonpb.Node, resource *resourcepb.
 	if resource != nil {
 		resourceType := resource.GetType()
 		if resourceType != "" {
-			jTags = append(jTags, &jaeger.Tag{
+			resourceTypeTag := &jaeger.Tag{
 				Key:   opencensusResourceType,
 				VType: jaeger.TagType_STRING,
 				VStr:  &resourceType,
-			})
+			}
+			jTags = append(jTags, resourceTypeTag)
 		}
 		for k, v := range resource.GetLabels() {
-			jTags = append(jTags, &jaeger.Tag{
+			str := v
+			resourceTag := &jaeger.Tag{
 				Key:   k,
 				VType: jaeger.TagType_STRING,
-				VStr:  &v,
-			})
+				VStr:  &str,
```

Comment:
So this is the bug :-)

OK, I see now, the intent of other fixes to establish a more foolproof pattern that makes it slightly more difficult to accidentally use a pointer to a loop variable.

LGTM.

Reply:
wow, I am confused why can we not use pointer directly to the map value?

Reply:
If my understanding is correct the following happens:

In this case it would point to for loop variable `v` not map value. For loop variable gets a copy of map value (which happens to be a slice descriptor of the string). You can point to loop variable, but variable's value is updated on every loop iteration. The  consequence is that all `resourceTag` object's `VStr` fields points to the same local variable `v` (which obviously can contain only one value).
Doing `str := v` creates a new slice descriptor pointing to the string slice contained in the map, and new slice descriptor is copied into `str` variable on each iteration. The result is what we expected in the first place: VStr points to the correct string slice (different for each iteration).
I know it is confusing. I am myself not a big fun of how Go's syntax hides all these referential semantics of strings, slices and variables.